src/packet.c

```diff
@@ -405,6 +405,7 @@ packet_x11_open(LIBSSH2_SESSION * session, unsigned char *data,
             rc = _libssh2_transport_send(session, x11open_state->packet, 17,
                                          NULL, 0);
             if(rc == LIBSSH2_ERROR_EAGAIN) {
+                LIBSSH2_FREE(session, channel);
```

Comment:
If we free `channel` here, what happens when it comes back in to this this function after EAGAIN is hit and the state is currently set to `libssh2_NB_state_created` which assumes channel is already allocated and usable?

Reply:
Secondly, just freeing channel wouldn't be enough, you'd also need to free `channel->channel_type`. That said, I don't think this is a leak.

Reply:
Thanks @willco007!